languages/cpp/reference/exercise-concepts/reverse-string.md

```diff
@@ -0,0 +1,77 @@
+# Reverse String
+
+## Task
+
+Reverse a string
+
+For example:
+input: "cool"
+output: "looc"
+
+
+## Example
+
+`reverse_string.h`
+
+```cpp
+#if !defined(REVERSE_STRING_H)
+#define REVERSE_STRING_H
+
+#include <string>
+
+namespace reverse_string
+{
+
+std::string reverse_string(const std::string& text);
+
+}
+
+#endif
+```
+
+`reverse_string.cpp`
+```cpp
+#include "reverse_string.h"
+
+namespace reverse_string
+{
+
+std::string reverse_string(const std::string& text)
+{
+    return std::string(text.rbegin(), text.rend());
+}
+
+}
+```
+
+## Concepts
+
+- .h file: the tests include `"reverse_string.h"`
+- .cpp file:  the most common approach is to use a `.cpp` for the definitions
+- namespaces: the function is member of a namespace
+- functions: used as the main entry point for the exercise
+- function arguments: the function has two arguments
+- strings: input is a string (or maybe string_view)
+- return values: the result is returned by the function
+- pass by reference to const: avoid making copy of input
```

Comment:
References should be an independent concept (in addition to that)

Reply:
Agree, I'll add that.